infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -494,44 +493,49 @@ private void registerToLogReplicationLock() {
-            case SOURCE:
-                log.info("Start as Source (sender/replicator)");
-                if (replicationManager == null) {
-                    replicationManager = new CorfuReplicationManager(replicationContext, localNodeDescriptor,
+        if (!topologyDescriptor.getRemoteSinkClusters().isEmpty()) {
+            log.info("Start as Source (sender/replicator)");
+            if (replicationManager == null) {
+                replicationManager = new CorfuReplicationManager(replicationContext, localNodeDescriptor,
                         remoteSessionToMetadataManagerMap, serverContext.getPluginConfigFilePath(), getCorfuRuntime(),
                         replicationConfigManager);
-                } else {
-                    // Replication Context contains the topology which
-                    // must be updated if it has changed
-                    replicationManager.setContext(replicationContext);
-                }
-                replicationManager.start();
-
-                // Set initial/default replication status for newly added Sink clusters
-                initReplicationStatusForRemoteClusters(true);
-                lockAcquireSample = recordLockAcquire(localClusterDescriptor.getRole());
-                processCountOnLockAcquire(localClusterDescriptor.getRole());
-                break;
-            case SINK:
-                log.info("Start as Sink (receiver)");
-
-                // Sink Site : the LogReplicationServer (server handler) will
-                // reset the LogReplicationSinkManager on acquiring leadership
-                interClusterServerNode.setLeadership(true);
-                lockAcquireSample = recordLockAcquire(localClusterDescriptor.getRole());
+            } else {
+                // Replication Context contains the topology which
+                // must be updated if it has changed
+                replicationManager.setContext(replicationContext);
+            }
+            replicationManager.start(fetchConnectionEndpoints());
 
-                // Set initial/default replication status for newly added Source clusters
-                initReplicationStatusForRemoteClusters(false);
-                processCountOnLockAcquire(localClusterDescriptor.getRole());
-                break;
-            default:
-                log.error("Log Replication not started on this cluster. Leader node {} id {} belongs to cluster with {} role.",
-                        localEndpoint, localNodeId, localClusterDescriptor.getRole());
-                break;
+            // Set initial/default replication status for newly added Sink clusters
+            initReplicationStatusForRemoteClusters(true);
+            lockAcquireSample = recordLockAcquire(localClusterDescriptor.getRole());
+            processCountOnLockAcquire(ClusterRole.SOURCE);
+        } else if (!topologyDescriptor.getRemoteSourceClusters().isEmpty()) {
+            log.info("Start as Sink (receiver)");
```

Comment:
It will be good to log the remote cluster id and replication model here, i.e., "Start as Sink for remote cluster X, Replication Model Y..'

Reply:
I'll add a TODO about this because this part is going to change in the connection model PR. That has the logic where a cluster can be both SOURCE/SINK.